frontend/src/components/CollapseButton.tsx

```diff
@@ -15,7 +15,7 @@
  */
 
 import * as React from 'react';
-import { CompareState } from '../pages/Compare';
+import { CompareState } from '../pages/CompareV1';
```

Comment:
Should I move this `CompareState` interface to the new `Compare.tsx` page? I can do that if we don't want any references to the old `V1` files.

Reply:
Good question: I think it is fine to keep the `CompareState` in `CompareV1`. The more important issue is that `CollapseButton` shouldn't import `CompareState`, do you have an idea how to better design `CollapseButton`?

Reply:
Investigating this file, I found that `CollapseButton` is only used for the `CompareV1.tsx` page, and uses that `CompareState` interface to define the `compareSetState` function that is passed in which dictates which components of the original page are collapsed (it is currently not a generalized component).

I think the simplest way to remove this import would be to have `CollapseButton` accept a function which calls the `setState` function, so you wouldn't need the `CompareState` type definition. This should generalize the component so it can be used on other pages with other type definitions as well.

I implemented this idea in the latest commit: https://github.com/kubeflow/pipelines/pull/7793/commits/d4bfd14d8e86237ef9b84dd404033f42ca9b122e. It's also shown working in the attached screencast:

https://user-images.githubusercontent.com/7987279/170638470-6ce62fbd-84ce-4e37-a87e-eb033192cb26.mp4

If there are other implementation ideas though, or if this should be separated into another PR, please let me know.

Reply:
It is awesome approach and you have nicely done the refactoring to generalize the Collapse button! I also appreciate that you have explained your approach using commits and video. Thank you Chris!